packages/keyring-controller/src/KeyringController.ts

```diff
@@ -240,12 +272,15 @@ export class KeyringController extends BaseControllerV2<
       },
     });
 
-    this.#keyring = new EthKeyringController(
-      Object.assign(
-        { initState: state },
-        { encryptor, keyringBuilders, cacheEncryptionKey },
-      ),
-    );
+    this.#keyring = new EthKeyringController({
+      // @ts-expect-error @metamask/eth-keyring-controller uses initState to
+      // initialize the persistent store, which is an object
+      // with a `vault` property
+      initState: state,
```

Comment:
Hmm, the `@metamask/eth-keyring-controller` types seem very broken here.

Reply:
Yes, we don't really initialise the state but just the vault

Reply:
Ah... it seems that since the types for `obs-store` are being defined simply as `declare module 'obs-store';`, any part of the interface ends up being `any`. That `any` trickles down to `this.store.getState().vault` on [this line](https://github.com/MetaMask/KeyringController/blob/2adb4323050d11fee7a34f974e679ba879c64284/src/KeyringController.ts#L236).

That said, it seems that we are using `obs-store` 4.x in `eth-keyring-controller`, but it's been converted to TypeScript, so we should not need to provide our own types if we upgrade to the latest version. Is that something you think would be worth doing?

Reply:
I think we have `@metamask/obs-store` which is shipped with types

Moreover, [this property](https://github.com/MetaMask/KeyringController/blob/2adb4323050d11fee7a34f974e679ba879c64284/src/types.ts#L10) should not be the entire state, as it would intend that passing anything other than the vault would be reflected on the initial state - but it isn't, because we never use it to initialize the rest of the state

Reply:
I created [this PR](https://github.com/MetaMask/KeyringController/pull/247) to update `obs-store` and fix discrepancies with types - it is breaking. 
We can decide if we want to wait for that